dtde/spark/cli/cluster_add_user.py

```diff
@@ -10,7 +9,7 @@ def setup_parser(parser: argparse.ArgumentParser):
                         help='The usernameto access your spark cluster\'s head node')
 
     auth_group = parser.add_mutually_exclusive_group()
-    auth_group.add_argument('-p', '--password',
+    auth_group.add_argument('-p', '--password', action=cli.Password, nargs='?', dest='password',
```

Comment:
I think doing it this way is breaking the usage of a ssh public key as it will ask for it all the time even if you have ssh_key passed as an argument or in the secrets.yml